manifests/init.pp

```diff
@@ -229,53 +254,53 @@
-  $log_timestamp                       = $::corosync::params::log_timestamp,
-  $log_file                            = $::corosync::params::log_file,
-  $log_file_name                       = $::corosync::params::log_file_name,
-  $debug                               = $::corosync::params::debug,
-  $log_stderr                          = $::corosync::params::log_stderr,
-  $syslog_priority                     = $::corosync::params::syslog_priority,
-  $log_function_name                   = $::corosync::params::log_function_name,
-  $rrp_mode                            = $::corosync::params::rrp_mode,
-  $netmtu                              = $::corosync::params::netmtu,
-  $ttl                                 = $::corosync::params::ttl,
-  $vsftype                             = $::corosync::params::vsftype,
-  $package_corosync                    = $::corosync::params::package_corosync,
-  $package_crmsh                       = $::corosync::params::package_crmsh,
-  $package_pacemaker                   = $::corosync::params::package_pacemaker,
-  $package_pcs                         = $::corosync::params::package_pcs,
-  $packageopts_corosync                = $::corosync::params::package_install_options,
-  $packageopts_pacemaker               = $::corosync::params::package_install_options,
-  $packageopts_crmsh                   = $::corosync::params::package_install_options,
-  $packageopts_pcs                     = $::corosync::params::package_install_options,
-  $version_corosync                    = $::corosync::params::version_corosync,
-  $version_crmsh                       = $::corosync::params::version_crmsh,
-  $version_pacemaker                   = $::corosync::params::version_pacemaker,
-  $version_pcs                         = $::corosync::params::version_pcs,
-  $set_votequorum                      = $::corosync::params::set_votequorum,
-  $votequorum_expected_votes           = $::corosync::params::votequorum_expected_votes,
-  $quorum_members                      = ['localhost'],
-  $quorum_members_ids                  = undef,
-  $token                               = $::corosync::params::token,
-  $token_retransmits_before_loss_const = $::corosync::params::token_retransmits_before_loss_const,
-  $compatibility                       = $::corosync::params::compatibility,
-  $manage_pacemaker_service            = $::corosync::params::manage_pacemaker_service,
-  $manage_pcsd_service                 = false,
-  $cluster_name                        = $::corosync::params::cluster_name,
-  $join                                = $::corosync::params::join,
-  $consensus                           = $::corosync::params::consensus,
-  $max_messages                        = $::corosync::params::max_messages,
-  $test_corosync_config                = $::corosync::params::test_corosync_config,
+  Enum['on', 'off'] $enable_secauth                                           = $::corosync::params::enable_secauth,
```

Comment:
Is there a reason this shouldn't continue to be a Boolean? You'd then have to do some sort of translation to make the param show up correctly in the config file, I can see both sides of this.

Reply:
> You'd then have to do some sort of translation to make the param show up correctly in the config file

I wanted to remove this translation. `corosync.conf` expects 'on' / 'off' (see [line 40](https://github.com/voxpupuli/puppet-corosync/pull/401/files#diff-ebc95a4d5f063d6c0899da029e5f12de)).
```
-  secauth:                             <%= @enable_secauth_real %>
+  secauth:                             <%= @enable_secauth %>
```

The question is: keep `enable_secauth_real` or break configurations?

I don't know which decision led to `enable_secauth_real` - it was present in the initial commit -, but other `_real` variables were removed in 2014 ([ab3b1847](https://github.com/voxpupuli/puppet-corosync/commit/ab3b1847)).

Reply:
As far as the interface to end-users, I think it's still cleaner to keep the parameter a bool, but I don't have a strong feeling about it.
BTW, I think if we do create a new var, it should be updated to `_enable_secauth` vs `enable_secauth_real`, which seems to be the better practice these days.

Reply:
A boolean makes more sense. Moreover, we can expect end-users to use a boolean in their existing configuration since this is the example given in the README.

IMHO, if the decision is to move forward with 'on/off', it would at least require a deprecation period with an update of the README and a notify for those still using a boolean. Thus it would need a specific type allowing both Enum and Boolean.

Also, I agree with @juniorsysadmin comment in #415:
> do data types first in one PR, then do boolean/string changes in another one afterwards.